### PR TITLE
replay: Add support to resume interrupted/crashed job [v2]

### PR DESCRIPTION
**Pull request by ldoktor**
This PR fixes the way we get info about not-executed-tests from "results.json" and then adds support to resume job by using `--replay-resume` argument. This is basically just a symlink to `--replay-test-status INTERRUPTED` and it can be also combined with other statuses. The only drawback could be when the "results.json" is not yet available (host crash, kill -9, ...) the "results.tap" is used and it does not support all test statuses and for the replay purpose it maps all failures to ERRORs.

v1: https://github.com/avocado-framework/avocado/pull/1942

Changes:
```yaml
v2: Add documentation
v2: Add commit to always sync basic job-related data on disk
```

### CODE REVIEW

**Comment by apahim** [comment]
Ok, I like this version a lot. I will run some tests in a few moments, but I have a thought I'd like to share in advance. Picture a test that is causing the crash... should we replay that test? I'm considering a scenario where user has an automated way to resume the job in case of crash. A test causing the crash would make the system to stay in a crash loop forever. Maybe we should add an option to skip the test that was under execution when the job was interrupted. What do you think?

**Comment by ldoktor** [comment]
I believe that is a corner case which shouldn't complicate the usage, but we do have the knowledge so we can improve it on such user request.

File: selftests/unit/test_replay.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
```
Comment (clebergnu) [inline selftests/unit/test_replay.py]: No need for a shebang, see d5fd51cf8cd69d19154d7c6ff3aa2f1513eea41a.

Comment (clebergnu) [inline selftests/unit/test_replay.py]: BTW, I cleaned the last few on #1969.

Comment (ldoktor) [inline selftests/unit/test_replay.py]: OK, I copy&pasted it from the remaining ones 😄
